mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature

```diff
@@ -5,51 +5,411 @@ Feature: Source-Record-Storage
+    Then status 201
+    #   Create record 1
+    Given path 'source-storage','records'
+    And request record1
+    When method POST
+    Then status 201
+    * record1.order = 2
+    #   Update record
+    Given path 'source-storage', 'records', recordId
+    And request record1
+    When method PUT
+    Then status 200
+    #   Get record
+    Given path 'source-storage', 'records', recordId
+    When method GET
+    Then status 200
+    And assert response.order == 2
 
-  @Undefined
   Scenario: Test filtering by record type
     * print 'Create snapshots, save records with different types linked to multiple snapshots, get collection filtered by record type'
+    * def snapshotId = uuid()
+    * def snapshot1 = read(snapshotPath)
+
+    * def recordId = uuid()
+    * def record1 = read(recordPath)
+
+    * def recordId = uuid()
+    * def record2 = read(recordPath)
+    * record2.recordType = 'MARC_HOLDING'
+
+    * def recordId = uuid()
+    * def record3 = read(recordPath)
+
+    * def snapshotId = uuid()
+    * def snapshot2 = read(snapshotPath)
+
+    * def recordId = uuid()
+    * def record4 = read(recordPath)
+    * record4.recordType = 'MARC_HOLDING'
+    #   Create snapshot 1
+    Given path 'source-storage','snapshots'
+    And request snapshot1
+    When method POST
+    Then status 201
+    #   Create snapshot 2
+    Given path 'source-storage','snapshots'
+    And request snapshot2
+    When method POST
+    Then status 201
+    #   Create record 1
+    Given path 'source-storage','records'
+    And request record1
+    When method POST
+    Then status 201
+    #   Create record 3
+    Given path 'source-storage','records'
+    And request record3
+    When method POST
+    Then status 201
+    #   Create record 2
+    Given path 'source-storage','records'
+    And request record2
+    When method POST
+    Then status 201
+    #   Create record 4
+    Given path 'source-storage','records'
+    And request record4
+    When method POST
+    Then status 201
+    #   Get records
+    Given path 'source-storage', 'records'
+    And param recordType = "MARC_HOLDING"
+    When method GET
+    Then status 200
+    And assert response.totalRecords == 2
 
-  @Undefined
   Scenario: Test creating error record if parsed content is invalid
     * print 'Create snapshot, try to save record with invalid parsed content - verify error record is created'
+    * def snapshotId = uuid()
+    * def snapshot1 = read(snapshotPath)
+
+    * def recordId = uuid()
+    * def record1 = read(errorRecordPath)
+    #   Create snapshot 1
+    Given path 'source-storage','snapshots'
+    And request snapshot1
+    When method POST
+    Then status 201
+    #   Create record 1
+    Given path 'source-storage','records'
+    And request record1
+    When method POST
+    Then status 201
+    # Get record by id
+    Given path 'source-storage', 'records', recordId
+    When method GET
+    Then status 200
+    And assert response.id == recordId
```

Comment:
please, add verification, according to test description, that error record is created
```suggestion
    And assert response.id == recordId
    And match response.errorRecord == '#notnull'
```
please, add verification, according to test description, that error record is created

Reply:
done